pkg/scheduler/framework/plugins/nodevolumelimits/csi.go

```diff
@@ -134,13 +138,28 @@ func (pl *CSILimits) Filter(ctx context.Context, _ *framework.CycleState, pod *v
 }
 
 func (pl *CSILimits) filterAttachableVolumes(
-	csiNode *storagev1.CSINode, volumes []v1.Volume, namespace string, result map[string]string) error {
+	pod *v1.Pod, csiNode *storagev1.CSINode, volumes []v1.Volume, namespace string, result map[string]string) error {
 	for _, vol := range volumes {
-		// CSI volumes can only be used as persistent volumes
-		if vol.PersistentVolumeClaim == nil {
+		// CSI volumes can only be used through a PVC.
+		pvcName := ""
+		ephemeral := false
+		switch {
+		case vol.PersistentVolumeClaim != nil:
+			pvcName = vol.PersistentVolumeClaim.ClaimName
+		case vol.Ephemeral != nil:
+			if !pl.enableGenericEphemeralVolume {
+				return fmt.Errorf(
+					"volume %s is a generic ephemeral volume, but that feature is disabled in kube-scheduler",
+					vol.Name,
+				)
+			}
+			// Generic ephemeral inline volumes also use a PVC,
+			// just with a computed name, and...
```

Comment:
and what? If this comment is continuing on L175 I would just say that here and say  the second part is checked later. If we end up adding more logic in between these lines, then this comment will be very confusing to follow.

Reply:
These two indeed belong together. I've made the clearer.

Reply:
Has this been updated?

Reply:
I'm sure I did, but I must have lost that change again during a rebase, or I never properly committed it - sorry!

One more time with feeling!

Reply:
I had only updated the comment in `non_csi.go`. Having two very similar files is annoying, but as I said in my other comment, I'd prefer to not turn this PR into a refactoring exercise of this filter.